pkgs/development/compilers/halide/default.nix

```
@@ -14,48 +15,81 @@
   openblas,
   blas,
   lapack,
+  removeReferencesTo,
+  ninja,
   pythonSupport ? false,
   python3Packages ? null,
+  wasmSupport ? false,
+  wabt,
+  doCheck ? true,
 }:
 
 assert blas.implementation == "openblas" && lapack.implementation == "openblas";
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "halide";
-  version = "18.0.0";
+  version = "19.0.0";
 
   src = fetchFromGitHub {
     owner = "halide";
     repo = "Halide";
-    rev = "v${version}";
-    hash = "sha256-BPalUh9EgdCqVaWC1HoreyyRcPQc4QMIYnLrRoNDDCI=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-0SFGX4G6UR8NS4UsdFOb99IBq2/hEkr/Cm2p6zkIh/8=";
   };
 
-  postPatch = ''
-    # See https://github.com/halide/Halide/issues/7785
-    substituteInPlace 'src/runtime/HalideRuntime.h' \
-      --replace '#if defined(__x86_64__) || defined(__i386__) || defined(__arm__) || defined(__aarch64__)
-    #define HALIDE_CPP_COMPILER_HAS_FLOAT16' \
-                '#if defined(__x86_64__) || defined(__i386__)
-    #define HALIDE_CPP_COMPILER_HAS_FLOAT16'
-  '';
+  patches = [
+    # The following two patches fix cmake/HalidePackageConfigHelpers.cmake to
+    # support specifying an absolute library install path (which is what Nix
+    # does when "lib" is included as a separate output)
+    (fetchpatch2 {
+      url = "https://github.com/halide/Halide/commit/ac2cd23951aff9ac3b765e51938f1e576f1f0ee9.diff";
+      hash = "sha256-sjio0L/ujgWzKQwqnFy6AZQntfb9iYi8vhixoBRMx5Y=";
```

```suggestion
      url = "https://github.com/halide/Halide/commit/ac2cd23951aff9ac3b765e51938f1e576f1f0ee9.diff?full_index=1";
      hash = "";
```
When using fetchpatch2 you must specify full_index=1 to ensure the hash does not change.

---

Ah thanks! I wondered why I got a hash mismatch error...